benchadapt/python/benchadapt/result.py

```diff
@@ -148,6 +151,7 @@ def _github_property(self, value: Optional[dict]):
         if value is None:
             value = _machine_info.detect_github_info()
         self._github_cache = value
+        self._maybe_set_run_name()
```

Comment:
A docstring on `_maybe_set_run_name()` might help, buuut I wonder: why does this not try to set a run name when `self.github.get("commit")` evaluates to `False`?

Reply:
Lovely. Thanks for the thoughtful reply. I am now slightly unconfused. :)

Reply:
To bring our various conversations together: technically, `run_name` is not required by the API (you can omit it from the payload) but right now the API doesn't like `"run_name": None`. I consider that a bug in the API.